hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java

```diff
@@ -43,17 +46,34 @@
 public class AvroKafkaSource extends AvroSource {
 
   private static final Logger LOG = LogManager.getLogger(AvroKafkaSource.class);
-
   private final KafkaOffsetGen offsetGen;
-
   private final HoodieDeltaStreamerMetrics metrics;
 
   public AvroKafkaSource(TypedProperties props, JavaSparkContext sparkContext, SparkSession sparkSession,
       SchemaProvider schemaProvider, HoodieDeltaStreamerMetrics metrics) {
     super(props, sparkContext, sparkSession, schemaProvider);
-    this.metrics = metrics;
+
     props.put("key.deserializer", StringDeserializer.class);
-    props.put("value.deserializer", KafkaAvroDeserializer.class);
+    String deserializerClassName = props.getString(DataSourceWriteOptions.KAFKA_AVRO_VALUE_DESERIALIZER(), "");
+
+    if (deserializerClassName.isEmpty()) {
+      props.put("value.deserializer", KafkaAvroDeserializer.class);
+    } else {
+      try {
+        if (schemaProvider == null) {
+          throw new HoodieIOException("SchemaProvider has to be set to use custom Deserializer");
+        }
+        props.put(DataSourceWriteOptions.SCHEMA_PROVIDER_CLASS_PROP(), schemaProvider.getClass().getName());
+        props.put(DataSourceWriteOptions.JAVA_SPARK_CONTEXT_PROP(), sparkContext);
+        props.put("value.deserializer", Class.forName(deserializerClassName));
```

Comment:
What is `value.deserializer` ? Is there a CONSTANT for this ?

Reply:
nope. but I can add one.